### PR TITLE
Added Otter Browser detection.

### PR DESCRIPTION
**Pull request by m4sk1n**
Signed-off-by: Marcin Mikołajczak <me@m4sk.in>

### CODE REVIEW

**Comment by fnando** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).
* 1 fixed issue! 🎉

You can see more details about this review at https://ebertapp.io/github/fnando/browser/pulls/310.

File: lib/browser/base.rb [hunk 1]
```diff
@@ -179,5 +179,10 @@ def proxy?
     def electron?(expected_version = nil)
       Electron.new(ua).match? && detect_version?(full_version, expected_version)
     end
+
```
Comment (fnando) [inline lib/browser/base.rb]: Trailing whitespace detected.

**Comment by fnando** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/fnando/browser/pulls/310.

File: lib/browser/otter-browser.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal:
```
Comment (fnando) [inline lib/browser/otter-browser.rb]: Trailing whitespace detected.

File: lib/browser/otter-browser.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: 
+
+module Browser
+  class OtterBrowser < Base
```
Comment (fnando) [inline lib/browser/otter-browser.rb]: Similar code found in 8 other locations (mass = 21) ([lib/browser/otter-browser.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/otter-browser.rb#L4-L18), [lib/browser/alipay.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/alipay.rb#L4-L18), [lib/browser/electron.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/electron.rb#L4-L19), [lib/browser/firefox.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/firefox.rb#L4-L18), [lib/browser/micro\_messenger.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/micro_messenger.rb#L4-L18), [lib/browser/nokia.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/nokia.rb#L4-L18), [lib/browser/phantom\_js.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/phantom_js.rb#L4-L18), [lib/browser/uc\_browser.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/uc_browser.rb#L4-L18), and [lib/browser/weibo.rb#4](https://github.com/fnando/browser/blob/9432836727430d1ea738f4ef36f565eb4225524c/lib/browser/weibo.rb#L4-L18))

**Comment by fnando** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/fnando/browser/pulls/310.

**Comment by fnando** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/fnando/browser/pulls/310.

**Comment by m4sk1n** [comment]
I don't understand it…

File: lib/browser/base.rb [hunk 4]
```diff
@@ -179,5 +179,10 @@ def proxy?
...
+    # Detect if browser is Otter Browser.
+    def otter_browser?(expected_version = nil)
+      OtterBrowser.new(ua).match? && detect_version?(full_version, expected_version)
```
Comment (fnando) [inline lib/browser/base.rb]: Line is too long. [84/80]

File: lib/browser/otter_browser.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Browser
+  class OtterBrowser < Base
```
Comment (fnando) [inline lib/browser/otter_browser.rb]: Similar code found in 8 other locations (mass = 21) ([lib/browser/otter\_browser.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/otter_browser.rb#L4-L18), [lib/browser/alipay.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/alipay.rb#L4-L18), [lib/browser/electron.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/electron.rb#L4-L19), [lib/browser/firefox.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/firefox.rb#L4-L18), [lib/browser/micro\_messenger.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/micro_messenger.rb#L4-L18), [lib/browser/nokia.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/nokia.rb#L4-L18), [lib/browser/phantom\_js.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/phantom_js.rb#L4-L18), [lib/browser/uc\_browser.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/uc_browser.rb#L4-L18), and [lib/browser/weibo.rb#4](https://github.com/fnando/browser/blob/73c4325b23ef6133120289c4faeef2b71430fbcf/lib/browser/weibo.rb#L4-L18))

**Comment by fnando** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 4 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/fnando/browser/pulls/310.

**Pull request closed without merging**